pkg/asset/installconfig/gcp/client.go

```diff
@@ -109,6 +110,42 @@ func (c *Client) GetPublicDomains(ctx context.Context, project string) ([]string
 	return publicZones, nil
 }
 
+// GetDNSZoneByName returns a DNS zone matching the `zoneName` if the DNS zone exists
+// and can be seen (correct permissions for a private zone) in the project.
+func (c *Client) GetDNSZoneByName(ctx context.Context, defaultProject, providedZoneName string) (*dns.ManagedZone, error) {
+	ctx, cancel := context.WithTimeout(context.TODO(), 1*time.Minute)
+	defer cancel()
+
+	project, zoneName, err := ParseZoneName(providedZoneName)
+	if err != nil {
+		return nil, err
+	} else if project == "" {
```

Comment:
There is no need to do an `else` here because the prior if is a return. You can flatten this and start a fresh `if project =="" {` on the next line.

```
if err != nil {
    return nil, err
}

if project == "" {
    project = defaultProject
}
```